src/Pim/Bundle/DataGridBundle/Datasource/Datasource.php

```diff
@@ -173,7 +173,7 @@ protected function getConfiguration($key, $isRequired = true)
             throw new \Exception(sprintf('"%s" expects to be configured with "%s"', get_class($this), $key));
         }
 
-        return $this->configuration[$key];
+        return isset($this->configuration[$key]) ? $this->configuration[$key] : null;
```

Comment:
If key does not exists, this creates a PHP Notice if array key does not exists, resulting in failure when in dev mode.